lib/babe/verify.go

```diff
@@ -128,45 +128,44 @@ func (v *VerificationManager) SetOnDisabled(index uint32, header *types.Header)
 // It checks the next epoch and config data stored in memory only if it cannot retrieve the data from database
 // It returns an error if the block is invalid.
 func (v *VerificationManager) VerifyBlock(header *types.Header) error {
-	var (
-		info *verifierInfo
-		has  bool
-	)
+	parentHeader, err := v.blockState.GetHeader(header.ParentHash)
+	if err != nil {
+		return fmt.Errorf("getting header: %w", err)
+	}
 
 	epoch, err := v.epochState.GetEpochForBlock(header)
 	if err != nil {
-		return fmt.Errorf("failed to get epoch for block header: %w", err)
+		return fmt.Errorf("getting epoch for block header: %w", err)
 	}
 
-	v.lock.Lock()
```

Comment:
don't need to lock anymore?

Reply:
Yeah, since this was protecting a map that was kinda replicating what we currently have on `dot/state/epoch.go`, also created a issue to cover another method `SetOnDisable` that should be removed and placed correctly in another place then we can fully get rid of these locks: https://github.com/ChainSafe/gossamer/issues/3930